control-plane/connect-inject/common/annotation_processor.go

```diff
@@ -0,0 +1,275 @@
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/constants"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v1alpha1"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ConsulNodeAddress = "127.0.0.1"
+)
+
+// PodAnnotationProcessor processes a pod annotation into pbmesh.Upstreams.
+type PodAnnotationProcessor struct {
+	enablePartitions bool
+	enableNamespaces bool
+}
+
+// NewPodAnnotationProcessor constructs a PodAnnotationProcessor for processing pod annotations into pbmesh.Upsreams.
+func NewPodAnnotationProcessor(enablePartitions, enableNamespaces bool) PodAnnotationProcessor {
+	return PodAnnotationProcessor{
+		enablePartitions: enablePartitions,
+		enableNamespaces: enableNamespaces,
+	}
+}
```

Comment:
~ I'm not sure this abstraction is super necessary since it only serves a single function call, unless it sets you up for more complex logic later - I might consider just adding this data as args to `ProcessUpstreams`